### ISSUE TITLE
Function literals in case expressions break vertical alignment of "case" 

### ISSUE THREAD

**Issue by fthomas** [scalameta/scalafmt#357]
Original:

``` scala
  def foo(x: Int) = x match {
    case 1 => (y: Int) => y
    case _ => (y: Int) => -y
  }
```

Scalafmt:

``` scala
  def foo(x: Int) = x match {
    case 1 =>
      (y: Int) =>
        y
      case _ =>
      (y: Int) =>
        -y
  }
```

I would like the input to look like this:

``` scala
  def foo(x: Int) = x match {
    case 1 => (y: Int) => y
    case _ => (y: Int) => -y
  }
```

Using:
- 0.2.10
- --style default

**Comment by olafurpg** [comment]
I see two issues: 1) the broken indentation and 2) that scalafmt forces a newline for the lambda `=>`. I've fixed the first issue in #358 but the second one will need a more careful look.

**Comment by olafurpg** [comment]
![hqdefault](https://user-images.githubusercontent.com/1408093/39955615-097d2646-55d2-11e8-87ed-92bb31dfb6a9.jpg)

This got fixed in #1145 by upgrading to Scalameta v1.8. I can't explain exactly why it got fixed, but I ran scalafmt on ~2 million lines of code to check it and indeed a ton of small lambda bodies got moved onto the same line as `=>` as expected.
